swap/swap.go

```diff
@@ -80,19 +82,46 @@ func NewParams() *Params {
 }
 
 // New - swap constructor
-func New(stateStore state.Store, prvkey *ecdsa.PrivateKey, backend contract.Backend) *Swap {
+func new(stateStore state.Store, prvkey *ecdsa.PrivateKey, backend contract.Backend, disconnectThreshold uint64, paymentThreshold uint64) *Swap {
 	return &Swap{
 		store:               stateStore,
 		peers:               make(map[enode.ID]*Peer),
 		backend:             backend,
 		owner:               createOwner(prvkey),
 		params:              NewParams(),
-		paymentThreshold:    DefaultPaymentThreshold,
-		disconnectThreshold: DefaultDisconnectThreshold,
-		oracle:              NewPriceOracle(),
+		disconnectThreshold: disconnectThreshold,
+		paymentThreshold:    paymentThreshold,
+		honeyPriceOracle:    NewHoneyPriceOracle(),
 	}
 }
 
+// NewSWAP - swap constructor with integrity checks
+func NewSWAP(dbPath string, prvkey *ecdsa.PrivateKey, backendURL string, disconnectThreshold uint64, paymentThreshold uint64) (*Swap, error) {
+	// we MUST have a backend
+	if backendURL == "" {
+		return nil, errors.New("swap init error: no backend URL given")
+	}
+	log.Info("connecting to SWAP API", "url", backendURL)
+	// initialize the balances store
+	stateStore, err := state.NewDBStore(filepath.Join(dbPath, "swap.db"))
+	if err != nil {
+		return nil, fmt.Errorf("swap init error: %s", err)
+	}
+	if disconnectThreshold < paymentThreshold {
```

Comment:
I think it should be `<=`

Reply:
agreed, but also change the message accordingly if you apply this change

Reply:
Agree and changed